.specify/memory/constitution.md

```diff
@@ -0,0 +1,139 @@
+<!--
+Sync Impact Report:
+Version Change: N/A → 1.0.0
+Initial Constitution Creation
+Added Sections:
+  - Core Principles (6 principles)
+  - Data Management Standards
+  - Development Workflow
+  - Governance
+Templates Requiring Updates:
+  ✅ .specify/templates/plan-template.md - to be validated
+  ✅ .specify/templates/spec-template.md - to be validated
+  ✅ .specify/templates/tasks-template.md - to be validated
+Follow-up TODOs: None
+-->
+
+# Agentic Neurodata Conversion Constitution
+
+## Core Principles
+
+### I. MCP-Centric Architecture
+
+All functionality MUST be exposed through standardized MCP (Model Context Protocol) tools. Direct agent invocation from external clients is prohibited. The MCP server acts as the single orchestration layer, ensuring consistent interfaces and enabling horizontal scaling.
+
+**Rationale**: Centralizing all operations through MCP tools creates a unified API surface, simplifies client integration, and allows independent scaling of internal agents.
+
+### II. Agent Specialization
+
+Each internal agent (Conversation, Conversion, Evaluation, Knowledge Graph) MUST handle a single domain of responsibility. Cross-domain operations require coordination through the MCP server, not direct agent-to-agent communication.
+
+**Rationale**: Clear separation of concerns ensures maintainability, testability, and allows independent evolution of agent capabilities.
```

Comment:
we should discuss this. It seems to me like the mcp server should act as the clean interface that abstracts the structure of the underlying solution into a single clean interface. Having the agents call the server themselves results in a circular logic that may be too difficult to track. I imagine I am missing some key details.